src/conduits/java/com/enderio/conduits/common/blocks/ConduitBlock.java

```diff
@@ -241,7 +244,9 @@ public static void handleShiftYeta(PlayerInteractEvent.RightClickBlock event) {
                 @Nullable
                 IConduitType<?> type = conduit.getShape().getConduit(event.getPos(), event.getHitVec());
                 if (type != null) {
-                    conduit.removeTypeAndDelete(type, true);
+                    List<ItemStack> drops = Collections.singletonList(new ItemStack(type.getConduitItem()));
+                    conduit.removeTypeAndDelete(type, false);
+                    PlayerInteractionUtil.putStacksInInventoryFromWorldInteraction(event.getEntity(), event.getPos(), drops);
```

Comment:
Instead of Collections#singletonList use List#of instead.
conduit#removeTypeAndDelete has only a single usage (this one, remove the parameter that is always false and pass false in the method that used this parameter

Reply:
@agnor99 What do you think about creating a new method in the PlayerInteractionUtil that does the same thing but just takes a single ItemStack instead of a list? That way we don't have to create a list to begin with. I think putting a single item into the players inventory could be a common need. It would look like this:

```
     /**
     * Tries to transfer a single ItemStack, which was the result of a world interaction, into the player's inventory, and if not successful,
     * drops them on the ground where the event took place.
     *
     * @param player - the player that performed the event.
     * @param interactionPosition - Where the interaction took place, i.e. where the player clicked, the block was etc.
     * @param item - The ItemStack that comes from the event.
     */
    public static void putItemInInventoryFromWorldInteraction(Player player, BlockPos interactionPosition, ItemStack item){
        Level level = player.level();
        if(!player.addItem(item)){
            level.addFreshEntity(new ItemEntity(level, interactionPosition.getX()+0.5, interactionPosition.getY()+0.5, interactionPosition.getZ()+0.5, item));
        }
    }
```

Reply:
looks good, you can add it like that

Reply:
Sweet! I will need a little clarification on your request for conduit#removeTypeAndDelete. Are you asking for me to remote the boolean shouldDrop parameter all together since it's always false?

Reply:
yep, exactly that